lib/ansible/plugins/filter/core.py

```diff
@@ -565,6 +565,16 @@ def path_join(paths):
         raise AnsibleFilterTypeError("|path_join expects string or sequence, got %s instead." % type(paths))
 
 
+def commonpath(paths):
+    '''
+    takes a list of paths and it returns the longest common path from the given list.
+    '''
```

Comment:
I understand that this has been copied from the code above, but those aren't the best examples of writing docstrings, unfortunately.

Per [PEP 257](https://peps.python.org/pep-0257/), the first line should be an imperative-style active-voice sentence. Starting with a capital letter, ending with a trailing period, fitting in 72 chars. If it doesn't, maybe it's too wordy and some of it needs to elsewhere.
If there's important details to add, those should go into an extended description section in a separate paragraph, separated with one empty line after the title.

Here's one way of solving this:
```suggestion
    """Retrieve the longest common path from the given list.

    :param paths: A list of file system paths.
    :type paths: List[str]

    :returns: The longest common path.
    :rtype: str
    """
```

(the param definitions here follow Sphinx's markup declared @ https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#info-field-lists)

Reply:
No problem, I thought we aren't using descriptive docstrings for the functions here, thus copied the same as above to keep the singularity in code. Will make this change.